app/components/post_list/post/body/content/permalink_preview/index.tsx

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {withDatabase, withObservables} from '@nozbe/watermelondb/react';
+import {of as of$} from 'rxjs';
+import {switchMap, distinctUntilChanged} from 'rxjs/operators';
+
+import {fetchUsersByIds} from '@actions/remote/user';
+import {withServerUrl} from '@context/server';
+import {observePost} from '@queries/servers/post';
+import {observeConfigBooleanValue} from '@queries/servers/system';
+import {observeUserOrFetch, observeTeammateNameDisplay, observeCurrentUser} from '@queries/servers/user';
+
+import PermalinkPreview from './permalink_preview';
+
+import type {WithDatabaseArgs} from '@typings/database/database';
+
+const enhance = withObservables(['embedData', 'serverUrl'], ({database, embedData, serverUrl}: WithDatabaseArgs & {embedData: PermalinkEmbedData; serverUrl?: string}) => {
+    const showPermalinkPreviews = observeConfigBooleanValue(database, 'EnablePermalinkPreviews', false);
+    const teammateNameDisplay = observeTeammateNameDisplay(database);
+
+    const userId = embedData?.post?.user_id;
+    const author = userId ? observeUserOrFetch(database, serverUrl || '', userId, fetchUsersByIds) : of$(undefined);
```

Comment:
this should be only the observer and in the component you could call the fetch function if the author is undefined. Also this brings the question, what happens if the author is deactivated?

I've added a comment in the `observeUserOrFetch` function prior to this comment.

Additionally, serverUrl should be defined here and not optional, but by removing this function and only observe you don't even need the `serverUrl` here

Reply:
True I have updated the code keeping in mind your previous comments. Regarding the deactivated user here what will happen: 

- Deactivated users are still fetched by fetchUsersByIds - the API returns them
- User data is stored in the local database (including delete_at timestamp)
- DisplayUsername works normally - shows the user's actual name/username
- Profile picture displays - shows the user's actual profile picture
- No visual indication that the user is deactivated in permalink previews